src/tests/ftest/util/server_utils.py

```diff
@@ -48,6 +53,193 @@ class ServerFailed(Exception):
+            uri (str, optional): path to uri file. Defaults to None.
+            env (dict, optional): dictionary on env variable names and values.
+                Defaults to None.
+
+        Raises:
+            ServerFailed: if the specified job manager is unsupported.
+
+        Returns:
+            str: daos_sever launch command
+
+        """
+        if manager is None:
+            # Run locally
+            return self.__str__()
+
+        elif manager.endswith("orterun"):
+            if env:
+                assign_env = [
+                    "{}={}".format(key, val) for key, val in env.items()]
+            args = [
+                "-np {}".format(len(self.hosts)),
+                "-hostfile {}".format(self.hostfile),
+                "--enable-recovery",
+            ]
+            args.extend(["-x {}".format(assign) for assign in assign_env])
+            if uri is not None:
+                args.append("--report-uri {}".format(uri))
+
+        else:
+            raise ServerFailed("Unsupported job manager: {}".format(manager))
+
+        return "{} {} {}".format(manager, " ".join(args), self.__str__())
+
+    def start(self, manager, verbose=True, env=None, timeout=600):
+        """Start the daos server on each specified host.
+
+        Args:
+            manager (str): mpi job manager command
+            verbose (bool, optional): [description]. Defaults to True.
+            env ([type], optional): [description]. Defaults to None.
+            timeout (int, optional): [description]. Defaults to 600.
+
+        Raises:
+            ServerFailed: if there are issues starting the servers
+
+        """
+        if self.process is None:
+            # Start the daos server as a subprocess
+            kwargs = {
+                "cmd": self.get_launch_command(manager),
+                "verbose": verbose,
+                "allow_output_check": "combined",
+                "shell": True,
+                "env": env,
+                "sudo": True,
+            }
+            self.process = process.SubProcess(**kwargs)
+            self.process.start()
+
+            # Wait for 'DAOS I/O server' messages to appear in the daos_server
+            # output indicating that the servers have started
+            start_time = time.time()
+            start_msgs = 0
+            timed_out = False
+            while start_msgs != len(self.hosts) and not timed_out:
+                output = self.process.get_stdout()
+                start_msgs = len(re.findall("DAOS I/O server", output))
+                timed_out = time.time() - start_time > timeout
+
+            if start_msgs != len(self.hosts):
+                err_msg = "{} detected starting {}/{} servers".format(
+                    "Timed out" if timed_out else "Error",
+                    start_msgs, len(self.hosts))
+                print("{}:\n{}".format(err_msg, self.process.get_stdout()))
+                raise ServerFailed(err_msg)
+
+    def stop(self):
```

Comment:
how is this different to Kill_server below?

Reply:
`kill_server()` is more abrupt. stop() tries to poll the process and sends it subsequent signals to stop the server in a cleaner way.